stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/TimeDecorator.kt

```diff
@@ -5,54 +5,66 @@ import androidx.core.view.isVisible
 import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyFileAttachmentsViewHolder
 import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyMediaAttachmentsViewHolder
 import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithFileAttachmentsViewHolder
 import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithMediaAttachmentsViewHolder
 import io.getstream.chat.android.ui.utils.extensions.getCreatedAtOrNull
+import io.getstream.chat.android.ui.utils.extensions.isEphemeral
 
 internal class TimeDecorator(private val dateFormatter: DateFormatter) : BaseDecorator() {
 
     override fun decoratePlainTextMessage(viewHolder: MessagePlainTextViewHolder, data: MessageListItem.MessageItem) {
-        setupTime(viewHolder.binding.tvTime, data)
+        setupMessageFooter(viewHolder.binding.messageFooter, data)
     }
 
     override fun decoratePlainTextWithMediaAttachmentsMessage(
         viewHolder: PlainTextWithMediaAttachmentsViewHolder,
         data: MessageListItem.MessageItem
     ) {
-        setupTime(viewHolder.binding.tvTime, data)
+        setupMessageFooter(viewHolder.binding.messageFooter, data)
     }
 
     override fun decorateOnlyMediaAttachmentsMessage(
         viewHolder: OnlyMediaAttachmentsViewHolder,
         data: MessageListItem.MessageItem
     ) {
-        setupTime(viewHolder.binding.tvTime, data)
+        setupMessageFooter(viewHolder.binding.messageFooter, data)
     }
 
     override fun decorateOnlyFileAttachmentsMessage(
         viewHolder: OnlyFileAttachmentsViewHolder,
         data: MessageListItem.MessageItem
     ) {
-        setupTime(viewHolder.binding.tvTime, data)
+        setupMessageFooter(viewHolder.binding.messageFooter, data)
     }
 
     override fun decoratePlainTextWithFileAttachmentsMessage(
         viewHolder: PlainTextWithFileAttachmentsViewHolder,
         data: MessageListItem.MessageItem
     ) {
-        setupTime(viewHolder.binding.tvTime, data)
+        setupMessageFooter(viewHolder.binding.messageFooter, data)
     }
 
-    private fun setupTime(textView: TextView, data: MessageListItem.MessageItem) {
+    private fun setupMessageFooter(textView: TextView, data: MessageListItem.MessageItem) {
         val createdAt = data.message.getCreatedAtOrNull()
         when {
             data.positions.contains(MessageListItem.Position.BOTTOM).not() || createdAt == null -> textView.isVisible = false
             else -> {
-                textView.isVisible = true
-                textView.text = dateFormatter.formatTime(createdAt)
+                textView.apply {
+                    isVisible = true
+                    val createdAtTime = dateFormatter.formatTime(createdAt)
+                    if (data.message.isEphemeral()) {
+                        val footerTemplate = context.getString(R.string.stream_ui_ephemeral_msg_footer_template)
+                        text = String.format(footerTemplate, createdAtTime)
```

Comment:
Wouldn't it make things easier if this footnote message was a separate view? For example, we could add spacing between it and the timestamp, and that sort of thing.